src/layouts/default.vue

```diff
@@ -30,6 +33,12 @@ export default {
     rdHeader: Header,
     TheTitle,
   },
+  async fetch() {
+    if (this.$config.featureDiagnostics) {
+      await this.$store.dispatch('credentials/fetchCredentials');
+      await this.$store.dispatch('diagnostics/fetchDiagnostics', this.credentials);
```

Comment:
jsdoc also does code decoration???

Reply:
TypeScript supports [parsing JSDoc](https://www.typescriptlang.org/docs/handbook/intro-to-js-ts.html#providing-type-hints-in-js-via-jsdoc) for type hints, as a workaround for projects not able to move to TypeScript immediately.

Reply:
I just did a simple `as` part. BTW, the above code is wrong because you're calling the credentials getter before the fetcher, and end up with an empty object.